### ISSUE TITLE
Why LUKS instead of .tar.gpg?

### ISSUE THREAD

**Issue by koddo** [drduh/YubiKey-Guide#446]
Any particular reason? LUKS seems error-prone compared to a single command of encrypting a folder.

Thank you for the guide, by the way.

**Comment by drduh** [comment]
Indeed, one could symmetrically encrypt the private key and backup contents with GPG (though I would still recommend using a different passphrase from identity). The resulting archive would be more portable and convenient to use.

However, portability and convenience of accessing backups is not a goal of this guide and, in a way, LUKS limits portability to encourage use of a secure-ish operating environment. I also suppose the argument could be made that using different encryption programs to secure key material is "better", but in practice this is likely not relevant. In other words, in terms of at-rest encryption protection, LUKS and GPG are likely equivalent.

I'm curious what you mean by LUKS being error-prone. If it truly is a hassle for readers, we could simplify the instructions and move LUKS to Optional Hardening. What do you think? Is there anything else to consider?

**Comment by koddo** [comment]
Maybe it's just me, but I never interact with LUKS directly, which means I don't know what all those commands mean, and there's a lot to learn already about gpg and yubikey-manager. Also, by error-prone I mean manipulation of partitions in `/dev` manually: commands like `sudo dd if=/dev/zero of=/dev/sdc bs=4M count=1` seem dangerous for sleep-deprived users, I didn't mess up purely by luck.

**Comment by drduh** [comment]
@forbytten your approach is novel and perhaps technically superior, but I understand the original request as wanting to reduce overall complexity during setup.

can we do this without also materially impacting the security and durability of materials?

**Comment by koddo** [comment]
@forbytten, this is a great summary, thank you.

A couple of questions: 

1. When the user encrypts and decrypts backups while running a linux live image, the risk of plaintext persistence is mitigated, as everything is on in-memory filesystem, right?

2. What is image exfiltration? And why the risks are higher in the case of tar+gpg? I couldn't find explanation myself.

**Comment by koddo** [comment]
@forbytten, thank you for the explanation and analysis.

**Comment by koddo** [comment]
@forbytten, if it's advisable to use gpg symmetric encryption with argon2id, a normal good passphrase could be just hashed using the argon2 command line utility before feeding it to gpg. If I understand this correctly.

Thanks for the suggestion though, TIL.

**Comment by koddo** [comment]
@forbytten, that's unfortunate metadata is so fragile. I wouldn't use it for anything that could ever leave ext3 filesystem.

Another option would be putting this argon2-parameters file alongside with gpg-encrypted data in a tar archive, this way it's not going to be lost. This is much more user-friendly, if we are designing a fool-proof way to make backups.

**Comment by koddo** [comment]
> high-entropy passphrase will be KDF'd into a 128-bit key

@dimitry12, as far as I understand, this only applies to asymmetric encryption, because gpg-agent protects private keys with only AES-128 for some reason. And the good news this doesn't concern us, because private keys never leave yubikeys.

And when we use gpg for symmetric encryption of backups, it does use the cypher specified by user. Which is AES-256, thanks to the hardened gpg.conf from this guide.

```
$ gpg -c test.txt
$ gpg -vv test.txt.asc
...
:symkey enc packet: version 4, cipher 9, aead 0,s2k 3, hash 10
...
```

`Cipher 9` here means AES-256, and `hash 10` is SHA-512, see rfc4880 [section 9.2](https://www.rfc-editor.org/rfc/rfc4880#section-9.2) and section 9.4.

Good news again, S2K doesn't reduce entropy, it preserves entropy of passphrase. Unfortunately though, it's bad for passphrase stretching. This is why argon2id is recommended instead. Or rather, in addition, as discussed above.

> added benefit of partition being unmounted after reboot 

LUKS+loop is a good choice though, especially when keys are generated in non-ephemeral environments.

**Comment by drduh** [comment]
@forbytten @koddo can we summarize this discussion in a `storage.md` or equivalent artifact?

**Comment by koddo** [comment]
@drduh, I'm not sure what you mean, but I'm not against this if it's helpful in any way.

**Comment by drduh** [comment]
I recommend sending a PR to create a new file in this repository (something like `storage.md`), starting with the above table comparing backup storage options. Then link this new file from the Backup keys section so users can select the best option individually.
